engines/stark/ui/menu/saveloadmenu.cpp

```diff
@@ -132,11 +159,13 @@ SaveDataWidget::SaveDataWidget(int slot, Gfx::Driver *gfx, SaveLoadMenuScreen *s
 		_screen(screen),
 		_thumbWidth(StarkUserInterface->kThumbnailWidth),
 		_thumbHeight(StarkUserInterface->kThumbnailHeight),
-		_isMouseHovered(false),
 		_texture(gfx->createTexture()),
 		_outline(gfx->createTexture()),
-		_surfaceRenderer(gfx->createSurfaceRenderer()) {
-	// Obtain the thumbnail
+		_surfaceRenderer(gfx->createSurfaceRenderer()),
+		_textDesc(gfx),
+		_textTime(gfx),
+		_isMouseHovered(false) {
+	// Load the corresponding save slot data
 	Common::String filename = StarkEngine::formatSaveName(ConfMan.getActiveDomainName().c_str(), _slot);
 	Common::InSaveFile *save = g_system->getSavefileManager()->openForLoading(filename);
 	if (save) {
```

Comment:
I feel like putting a `delete save;` at the end is needed, but it will crash the game. The `StarkEngine::loadGameState()` also doesn't delete the `save`, why is that?

Reply:
The constructor for StateReadStream() is two arguments, the file and a flag (DisposeAfterUse::YES and DisposeAfterUse::NO) the flag indicates if the input file should be deleted or not after use. 

In your case you used it with 1 argument which looks like that defaults to the 2 argument case with the flag set to dispose. I am not exactly sure where that is happening since the constructor to SubReadStream defaults the dispose to NO (see common/substeam.h). StateReadStream is a SeekableSubReadStream which is also a SubReadStream.

So you don't need to delete save since that is already being done by the destructor of StateReadStream, but just to be clear you may want to explicitly set the 2nd argument to be DisposeAfterUse::YES in the constructor of StateReadStream.

Alternately, you can set it to NO and delete the save manually.